SofaKernel/modules/SofaBaseTopology/src/SofaBaseTopology/EdgeSetGeometryAlgorithms.inl

```diff
@@ -899,5 +899,68 @@ void EdgeSetGeometryAlgorithms<DataTypes>::initPointAdded(PointID index, const c
     }
 }
 
+template<class DataTypes>
+bool EdgeSetGeometryAlgorithms<DataTypes>::computeEdgeSegmentIntersection(EdgeID edgeID,
+    const sofa::defaulttype::Vec<3, double>& a,
+    const sofa::defaulttype::Vec<3, double>& b,
+    double &baryCoef)
+    //intersection=edge[0]+baryCoef*(edge[1]-edge[0])
+{
+    double EPS = 1e-05;
+    bool is_intersect = false;
+    
+    const Edge& e = this->m_topology->getEdge(edgeID);
+    const VecCoord& p = (this->object->read(core::ConstVecCoordId::position())->getValue());
+    const typename DataTypes::Coord& c0 = p[e[0]];
+    const typename DataTypes::Coord& c1 = p[e[1]];
+    //sofa::defaulttype::Vec<3, Real> p0 = defaulttype::Vector3(DataTypes::getCPos(p[e[0]]));
+    //sofa::defaulttype::Vec<3, Real> p1 = defaulttype::Vector3(DataTypes::getCPos(p[e[1]]));
+    
+    sofa::defaulttype::Vec<3, Real> p0;
```

Comment:
1- use initializer_list (much more readable)
2- actually if you use Real instead of double (previous comment) you would not need to cast the value.
so it would lead to thisL
```c++
sofa::defaulttype::Vec<3, Real> p0 { c0[0], c0[1], c0[2] }
```